api/src/main/java/org/openmrs/module/eptsreports/reporting/cohort/definition/EptsQuarterlyCohortDefinition.java

```diff
@@ -55,19 +52,24 @@ public CohortDefinition getCohortDefinition() {
   }
 
   public void setCohortDefinition(CohortDefinition cohortDefinition) {
-    validateParameters(cohortDefinition, "onOrAfter", "onOrBefore");
-    validateParameters(cohortDefinition, "startDate", "endDate");
+    validateParameters(cohortDefinition);
     this.cohortDefinition = cohortDefinition;
   }
 
-  private void validateParameters(
-          CohortDefinition cohortDefinition, String onOrAfter, String onOrBefore) {
-    List<Parameter> parameters = new ArrayList<>();
-    parameters.add(new Parameter(onOrAfter, "After Date", Date.class));
-    parameters.add(new Parameter(onOrBefore, "Before Date", Date.class));
-    if (!cohortDefinition.getParameters().containsAll(parameters)) {
-      throw new IllegalArgumentException(
-              "CohortDefinition must have both " + onOrAfter + " and " + onOrBefore + " parameters");
+  private void validateParameters(CohortDefinition cohortDefinition) {
+    String[][] valid = {{"onOrAfter", "onOrBefore"}, {"startDate", "endDate"}};
+    List<Parameter> cohortParams = cohortDefinition.getParameters();
+    boolean contains;
+    for (String[] v : valid) {
+      List<Parameter> params = new ArrayList<>();
+      params.add(new Parameter(v[0], "Start Date", Date.class));
+      params.add(new Parameter(v[1], "Start Date", Date.class));
```

Comment:
Did this label supposed to be endDate?

Reply:
Yes. Although the labels don't matter here.

I'll change, but this is supposed to be part of #255.